lib/resque/worker.rb

```diff
@@ -353,8 +375,10 @@ def prune_dead_workers
     # Registers ourself as a worker. Useful when entering the worker
     # lifecycle on startup.
     def register_worker
-      redis.sadd(:workers, self)
-      started!
+      redis.pipelined do
```

Comment:
Similar-but-different to master's implementation [here](https://github.com/al2o3cr/resque/pull/2/files#diff-102541f6f2ba476fceb8e45917e63420R233)